integration-tests/src/test/scala/tech/beshu/ror/integration/plugin/DataStreamApiSuite.scala

```diff
@@ -0,0 +1,364 @@
+/*
+ *    This file is part of ReadonlyREST.
+ *
+ *    ReadonlyREST is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    ReadonlyREST is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with ReadonlyREST.  If not, see http://www.gnu.org/licenses/
+ */
+package tech.beshu.ror.integration.plugin
+
+import org.scalatest.BeforeAndAfterEach
+import org.scalatest.freespec.AnyFreeSpec
+import tech.beshu.ror.integration.suites.base.support.BaseSingleNodeEsClusterTest
+import tech.beshu.ror.integration.utils.ESVersionSupportForAnyFreeSpecLike
+import tech.beshu.ror.utils.containers.EsClusterProvider
+import tech.beshu.ror.utils.elasticsearch._
+import tech.beshu.ror.utils.misc.Version
+
+import java.time.Instant
+
+trait DataStreamApiSuite extends AnyFreeSpec
+  with BaseSingleNodeEsClusterTest
+  with ESVersionSupportForAnyFreeSpecLike
+  with BeforeAndAfterEach {
+  this: EsClusterProvider =>
+
+  override implicit val rorConfigFileName: String = "/data_stream_api/readonlyrest.yml"
+
+  private lazy val client = clients.head.adminClient
+  private lazy val adminDocumentManager = new DocumentManager(client, esVersionUsed)
+  private lazy val adminDataStreamManager = new DataStreamManager(client)
```

Comment:
I see one problem with these tests. We test the API as an admin user which has no indices restriction. It would be better if we can use user1 | user2 | user3. Eg we should not be able to create a datastream that has ".ds-data-stream-dev-0001" index underneath if the user has no proper pattern allowed (eg. .ds-data-stream-dev*). I see you prepared proper ROR config, but these users were not used in this suite.